runtime/src/accounts_db.rs

```diff
@@ -2332,15 +2339,42 @@ pub mod tests {
         accounts.store(current_slot, &[(&dummy_pubkey, &dummy_account)]);
         accounts.add_root(current_slot);
 
-        purge_zero_lamport_accounts(&accounts, current_slot);
-        let accounts = reconstruct_accounts_db_via_serialization(&accounts, current_slot);
+        let (accounts, skip_account_assertion) = f(accounts, current_slot);
+
+        assert_eq!(4, accounts.accounts_index.read().unwrap().roots.len());
+        if skip_account_assertion {
```

Comment:
The following assertion causes failure currently for test_accounts_purge_chained_purge_after_snapshot_restore.

This should be fixed by #8337